frontend/src/concepts/pipelines/content/createRun/contentSections/TriggerTypeField.tsx

```diff
@@ -50,15 +52,16 @@ const TriggerTypeField: React.FC<TriggerTypeFieldProps> = ({ data, onChange }) =
         <FormGroup label="Run every" data-testid="run-every-group">
           <Split hasGutter>
             <SplitItem>
-              <NumberInputWrapper
+              <NumberInput
                 min={1}
-                value={parseInt(data.value) || 1}
-                onChange={(value) =>
+                value={numberPart}
+                onChange={(newNumberPart) => {
+                  const updatedValue = `${newNumberPart.currentTarget.value}${unitPart}`;
                   onChange({
                     ...data,
-                    value: replaceNumericPartWithString(data.value, value ?? 0),
-                  })
-                }
+                    value: updatedValue,
```

Comment:
so what you can do is clean the value like this. it will convert form the scientific notation to regular number

```ts
value.toLocaleString('fullwide', {useGrouping:false})
```

Reply:
you are doing ti on the string. it needs to be on the number

```tsx
onChange({
                    ...data,
                    value:
                      numberPart.toLocaleString('fullwide', { useGrouping: false }) + newUnitPart,
                  });
                }}
```

Reply:
the example you're mentioning is for the changing the unitPart, i am not understanding why need to clean the scientific value over here? 

If i try to replicate this for the NumberInput field, it's not working.

Can you explain why the clean up is required, and what issue will persist if we continue with the existing solution?